proknow-sdk/Patient/Entities/DoseItem.cs

```diff
@@ -10,21 +10,41 @@ namespace ProKnow.Patient.Entities
     public class DoseItem : EntityItem
     {
         /// <summary>
-        /// Downloads this entity asynchronously as a DICOM object to the specified folder
+        /// Downloads this dose asynchronously as a DICOM object to the specified folder or file
         /// </summary>
-        /// <param name="folder">The full path to the destination root folder</param>
-        /// <returns>The full path to the destination folder (root or a sub-folder) to which the file(s) were downloaded</returns>
-        public override Task<string> DownloadAsync(string folder)
+        /// <param name="path">The full path to the destination folder or file</param>
+        /// <returns>The full path to the file to which the dose was downloaded</returns>
+        /// <remarks>
+        /// <list type="bullet">
+        /// <item>
+        /// <description>
+        /// If the provided path is an existing folder, the dose will be saved to a file named
+        /// RD.{SOP instance UID}.dcm.
+        /// </description>
+        /// </item>
+        /// <item>
+        /// <description>
+        /// If the provided path is not an existing folder, the dose will be saved to the provided path.
+        /// </description>
+        /// </item>
+        /// </list>
+        /// </remarks>
+        public override Task<string> DownloadAsync(string path)
         {
-            if (File.Exists(folder))
+            string file = null;
+            if (Directory.Exists(path))
             {
-                throw new ArgumentException($"The destination folder path '{folder}' is a path to an existing file.");
+                file = Path.Combine(path, $"RD.{Uid}.dcm");
             }
-            if (!Directory.Exists(folder))
+            else
             {
-                Directory.CreateDirectory(folder);
+                var parentDirectoryInfo = Directory.GetParent(path);
```

Comment:
In this branch, should you be checking that `path` is a valid path (e.g., doesn't contain invalid characters) or is that handled by the `Directory.GetParent` method?

Same for PlanItem.cs and StructureSetItem.cs...

Reply:
It does several checks, including invalid characters, etc., but I'll put a try-catch around it so that we provide better error messages.

Reply:
I just tested the error messages and they are good as is since they include the path and a sufficient description of the error.